src/OrleansRuntime/Core/Dispatcher.cs

```diff
@@ -370,7 +375,20 @@ public void HandleIncomingRequest(Message message, ActivationData targetActivati
         {
             lock (targetActivation)
             {
-                if (targetActivation.State == ActivationState.Invalid)
+                if (targetActivation.Grain.IsGrain)
+                {
+                    var request = ((InvokeMethodRequest)message.GetDeserializedBody(this.serializationManager));
```

Comment:
This kind of improvement would be easy if we had a type like Bond's `IBonded` for deferred deserialization